kernel/of/fdt.c

```diff
@@ -264,6 +348,48 @@ fdt_node_next_subnode(struct fdt* fdt, struct fdt_node* subnode)
     return NULL;
 }
 
+size_t
+fdt_node_get_parents(
+    struct fdt* fdt,
+    struct fdt_node* node,
+    struct fdt_node* parents[],
+    size_t parents_len
+)
+{
+    // We can only traverse the FLAT device tree in forward order, so this is going to
+    // be painful (Also a big reason we want to unflatten the tree ASAP once we have
+    // memory allocation)
+
+    int max_depth = fdt_max_depth(fdt);
+    if (max_depth < 0) {
+        // Corrupted FDT
+        return 0;
+    }
+
+    struct fdt_node* curr_branch[max_depth + 1]; // C99 feature
+
+    int depth = 0;
```

Comment:
I'll make `depth` a `word_t` when that branch gets merged in. It really should be signed though, because if the tree isn't formed correctly we'll want a negative depth we can check for easily, not an unsigned underflow

Reply:
I see. What about a ssize_t then?

Reply:
It just needs to handle the maximum depth of the device tree, it could easily be handled by a `uint8_t` but since we don't need a fixed size this is an example where `word_t` makes sense

Reply:
The real `libFDT` uses an `int` for this purpose in their basically equivalent function for context

Reply:
Sure then